x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_store_data_client.ts

```diff
@@ -521,4 +639,50 @@ export class EntityStoreDataClient {
       errors: updateErrors,
     };
   }
+
+  private log(
+    level: Exclude<keyof Logger, 'get' | 'log' | 'isLevelEnabled'>,
+    entityType: EntityType,
+    msg: string
+  ) {
+    this.options.logger[level](
+      `[Entity Engine] [entity.${entityType}] [namespace: ${this.options.namespace}] ${msg}`
+    );
+  }
+
+  private audit(
+    action: EntityEngineActions,
+    resource: EntityStoreResource,
+    entityType: EntityType,
+    msg: string,
+    error?: Error
+  ) {
+    // NOTE: Excluding errors, all auditing events are currently WRITE events, meaning the outcome is always UNKNOWN.
+    // This may change in the future, depending on the audit action.
+    const outcome = error ? AUDIT_OUTCOME.FAILURE : AUDIT_OUTCOME.UNKNOWN;
+
+    // QUESTION: For EXECUTE action: Maybe START is better: https://www.elastic.co/guide/en/ecs/8.11/ecs-allowed-values-event-type.html#ecs-event-type-start
```

Comment:
Should we clarify al questions before merging the PR? Otherwise, we will have inconsistent audit logs.